src/clj/rems/db/user_mappings.clj

```diff
@@ -0,0 +1,46 @@
+(ns rems.db.user-mappings
+  (:require [clojure.tools.logging :as log]
+            [rems.db.core :as db]
+            [rems.json :as json]
+            [medley.core :refer [deep-merge]]
+            [schema.coerce :as coerce]
+            [schema.core :as s]))
+
+(s/defschema UserMappingsDataDb
+  {:from s/Str
+   :from-value s/Str
+   :to-value s/Str})
+
+(def ^:private coerce-UserMappingsDataDb
+  (coerce/coercer! UserMappingsDataDb coerce/string-coercion-matcher))
+
+(defn- format-user-mappings [user-mappings]
+  (let [usermappingsdata (-> (json/parse-string (:usermappingsdata user-mappings))
+                             coerce-UserMappingsDataDb)
+        {:keys [from from-value to-value]} usermappingsdata]
+    {from {from-value to-value}}))
```

Comment:
Maybe the mapping object should be just what was in the database. And then the index structure we cache can be different? So return the coerced `usermappingsdata` as is here? Just seems a bit odd to have it like this.

Reply:
removed cache in favor of indexed db table (and query)